packages/notifi-dapp-example/src/components/TopicOptions.tsx

```diff
@@ -213,31 +254,57 @@ export const TopicOptions = <T extends TopicRowCategory>(
                   }
                   selectOrInputValue(evt.target.value);
                 }
+                if (evt.target.value === '') {
+                  return options.includes(selectedOption?.toString() ?? '')
+                    ? setCustomInput('')
+                    : setCustomInput(selectedOption?.toString() ?? '');
+                }
+                // Case#2: If input === existing selected option, set customInput to empty
+                if (
+                  options.includes(evt.target.value) &&
+                  selectedOption === evt.target.value
+                )
+                  return setCustomInput('');
+
+                // Case#3: Subscribe new value
+                const { value } = validateCustomInputRange(
+                  evt.target.value,
+                  props.userInputParam.customInputConstraints,
+                );
+                setCustomInput(value.toString());
+                selectOrInputValue(value);
               }}
-              placeholder={props.placeholder ?? 'Custom'}
+              placeholder={
+                props.userInputParam.kind === 'price'
+                  ? 'Enter Price'
+                  : customInputPlaceholder
+              }
               value={customInput}
               className={` ${
                 props.userInputParam.options[0]
                   ? 'w-[6rem] text-center'
-                  : 'w-full pl-3'
+                  : `w-full ${
+                      'prefix' in prefixAndSuffix && prefixAndSuffix.prefix
+                        ? 'pl-7'
+                        : 'pl-3'
+                    }`
               } h-12 bg-notifi-card-bg rounded-md mr-2 text-notifi-text ${
                 customInput && props.userInputParam.options[0]
                   ? 'selected text-white border border-notifi-tenant-brand-bg focus:outline-none'
                   : 'focus:outline-none focus:border-notifi-card-border'
               }`}
             />
-            {props.userInputParam.options[0] ? null : (
+            {props.userInputParam.options[0] ? (
+              customInput ? (
+                <div className="absolute right-4 bottom-3">
+                  {'suffix' in prefixAndSuffix && prefixAndSuffix.suffix}
+                </div>
+              ) : null
+            ) : (
               <div className="absolute right-3 bottom-3">
-                {'prefix' in prefixAndSuffix && prefixAndSuffix.prefix}
+                {'suffix' in prefixAndSuffix && prefixAndSuffix.suffix}
               </div>
```

Comment:
Does this two conditions duplicated?

Reply:
nope, one for with Threshold Option button, one for without Threshold Option button.

Reply:
I think we will eventually remove the condition `props.userInputParam.options[0]`. If it is not proper to do it now, shall we make a `TODO:` comment so with can easily understand this part and upcoming action.